tgui/packages/tgui/interfaces/space_heater.js

```diff
@@ -0,0 +1,275 @@
+      if (Math.random() > 0.3) {
+        out.push("ERROR ");
+      } else {
+        out.push("Error ");
+      }
+    }
+    return out.map((kill, index) => (
+      <Box inline preserveWhitespace color="red" fontSize={randInt(10, 25) + "px"} key={index}>{kill}</Box>
+    ));
+  }
+  for (let i = 0; i < number; i++) {
+    let num = Math.random();
+    if (num < 0.2) {
+      out.push(theme === "syndicate" ? "HOT! HOT! HOT! ": "so cold... ");
+    } else if (0.2 < num < 0.4) {
+      out.push(theme === "syndicate" ? "IM BURNING! ": "im freezing... ");
+    } else if (0.4 < num < 0.6) {
+      out.push(theme === "syndicate" ? "HELP! ": "im shivering... ");
+    } else if (0.6 < num < 0.8) {
+      out.push(theme === "syndicate" ? "MELTING! ": "its so cold... ");
+    } else {
+      out.push(theme === "syndicate" ? "OVERHEATING! ": "chilly... ");
+    }
+  }
+  return out.map((kill, index) => (
+    <Box inline preserveWhitespace fontSize={randInt(10, 16) + "px"} key={index}>{kill}</Box>
+  ));
+};
+
+export const space_heater = (props, context) => {
+  const { data } = useBackend(context);
+  const {
+    emagged,
+    on,
+    name,
+    cell,
+    cell_charge,
+    set_temperature,
+  } = data;
+  return (
+    <Window
+      title={Glitch_Text(emagged, name, 1)}
+      theme={Find_Theme(emagged, set_temperature, on)}
+      width={350}
+      height={250}>
+      <Window.Content>
+        {HVAC_Death(emagged, cell, cell_charge) && (
+          <Modal
+            Align="center"
+            width={30.5}
+            height={18}
+          >
+            <Box backgroundColor={Set_Color(set_temperature, "#001414")}>
+              <Icon name={"skull-crossbones"} size={15} />
+            </Box>
+          </Modal>
+        )}
+        <BatteryStatus />
+        <TemperatureRegulator />
+      </Window.Content>
+    </Window>
+  );
+};
+
+const BatteryStatus = (props, context) => {
+  const { data, act } = useBackend(context);
+  const {
+    emagged,
+    on,
+    cell,
+    cell_name,
+    cell_charge,
+    set_temperature,
+  } = data;
+  return (
+    <Section title={Glitch_Text(emagged, "Battery status", 2)} grow={0}>
+      <LabeledList>
+        <LabeledList.Item
+          label={Glitch_Text(emagged, "Cell", 1)}
+          verticalAlign={"middle"}
+        >
+          <Flex justify={"space-between"}>
+            <Flex.Item>
+              <Button
+                icon={Set_Icon(Find_Theme(emagged, set_temperature, on))}
+                color={cell !== null ? Set_Color(set_temperature, "green", on) : "blue"}
+                disabled={on && !emagged}
+                onClick={() => cell !== null ? act('cellremove'): act('cellinstall')}
+                bold>
+                {cell !== null ? Glitch_Text(emagged, cell_name, 2): Glitch_Text(emagged, "Insert power cell", 3)}
+              </Button>
+            </Flex.Item>
+            <Flex.Item>
+              <Button
+                color={Set_Color(set_temperature, "blue", on)}
```

Comment:
The colour changing here is a little odd, it should probably just match how the scrubber UI looks since they're very similar in function.

Reply:
Oh yeah, looking at the scrubber, portable air pump, gas canisters and such its definitely odd. I was basing it off of the APC off/on switch because of a nother persons suggestion. Will change it up tommorow.

Reply:
Made it a singular button from 2 individual, and aligned the color of on/off button to other machinery.